### PR TITLE
Migrate to Swan Lake implementation

### PR DESCRIPTION
**Pull request by hemikak**

### CODE REVIEW

File: src/azure.ad/basicauth_inbound_handler.bal [hunk 1]
```diff
@@ -87,11 +88,11 @@ public type InboundUserAuthenticatorProvider object {
             passwordGrantConfig.clientSecret = clientSecret;
         }
 
-        http:BearerAuthHandler bearerAuthHandler = getOutboundOAuth2BearerHandler(passwordGrantConfig);
-        auth:OutboundAuthProvider provider = <auth:OutboundAuthProvider>bearerAuthHandler.authProvider;
+        oauth2:OutboundOAuth2Provider oAuthProvider = getOutboundOAuth2Provider(passwordGrantConfig);
```
Comment (ldclakmal) [inline src/azure.ad/basicauth_inbound_handler.bal:91]:
```suggestion
        oauth2:OutboundOAuth2Provider oauth2Provider = getOutboundOAuth2Provider(passwordGrantConfig);
```

Comment (hemikak) [reply src/azure.ad/basicauth_inbound_handler.bal]: I have fixed this with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/5/commits/fcf2013c0a9b2ff857af7d5791ce11ee566c980c

File: src/azure.ad/types_error.bal [hunk 2]
```diff
@@ -14,17 +14,15 @@
 // specific language governing permissions and limitations
 // under the License.
 
-public const AD_CLIENT_ERROR = "{ballerinax/azure.ad}AdClientError";
-
-public type AdClientError error<string, Details>;
+public type AdClientError error;
```
Comment (chamil321) [inline src/azure.ad/types_error.bal:17]: To make the error a nominal type that works within Ballerina's structural type system, we should make the error `distinct`. Then user can perform type test to distinguish these errors from the generic error

```suggestion
public type AdClientError distinct error;
```

File: src/azure.ad/types_error.bal [hunk 3]
```diff
@@ -14,17 +14,15 @@
...
+public type InvalidPayloadError error;
```
Comment (chamil321) [inline src/azure.ad/types_error.bal:18]:
```suggestion
public type InvalidPayloadError distinct error;
```

File: src/azure.ad/types_error.bal [hunk 4]
```diff
@@ -14,17 +14,15 @@
...
+public type GraphAPIError error<GraphAPIErrorDetails>;
```
Comment (chamil321) [inline src/azure.ad/types_error.bal:19]:
```suggestion
public type GraphAPIError distinct error<GraphAPIErrorDetails>;
```

Comment (hemikak) [reply src/azure.ad/types_error.bal]: Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/5/commits/22d494bd94ccc89dd2912d76f3e26db93cc75af7

File: src/azure.ad/types_error.bal (hunk 3 above)
Comment (hemikak) [reply src/azure.ad/types_error.bal]: Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/5/commits/22d494bd94ccc89dd2912d76f3e26db93cc75af7

File: src/azure.ad/types_error.bal (hunk 2 above)
Comment (hemikak) [reply src/azure.ad/types_error.bal]: Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/5/commits/22d494bd94ccc89dd2912d76f3e26db93cc75af7

**Pull request merged**